src/go/cmd/models.go

```diff
@@ -3,62 +3,126 @@
+	CodeDiagnosticLevelFatal CodeDiagnosticLevel = 4
 )
 
-type Navigation struct {
-	Text         string             `json:"Text"`
-	NavigationId string             `json:"NavigationId"`
-	ChildItems   []Navigation       `json:"ChildItems"`
+type CodeFile struct {
+	CrossLanguagePackageID string           `json:"CrossLanguagePackageId,omitempty"`
+	Diagnostics            []CodeDiagnostic `json:"Diagnostics,omitempty"`
+	Language               string           `json:"Language,omitempty"`
+	// LanguageVariant should be "None" if set
+	LanguageVariant string           `json:"LanguageVariant,omitempty"`
+	Name            string           `json:"Name,omitempty"`
+	Navigation      []NavigationItem `json:"Navigation,omitempty"`
+	PackageName     string           `json:"PackageName"`
+	PackageVersion  string           `json:"PackageVersion"`
+	ParserVersion   string           `json:"ParserVersion"`
+	ReviewLines     []ReviewLine     `json:"ReviewLines,omitempty"`
+}
+
+type NavigationItem struct {
+	ChildItems   []NavigationItem   `json:"ChildItems"`
+	NavigationID string             `json:"NavigationId"`
 	Tags         *map[string]string `json:"Tags"`
+	Text         string             `json:"Text"`
+}
+
+type ReviewLine struct {
+	// Children of the ReviewLine such as types in a namespace or methods on a type
+	Children        []ReviewLine `json:"Children,omitempty"`
+	CrossLanguageID string       `json:"CrossLanguageId,omitempty"`
+	// IsContextEndLine sets this line as the end of a context e.g. a line with token "}" or an empty line after a type definition
+	IsContextEndLine bool `json:"IsContextEndLine,omitempty"`
+	// IsHidden hides lines that shouldn't be visible by default
+	IsHidden bool `json:"IsHidden,omitempty"`
+	// LineID is for lines that should allow comments. Its value must be unique within the CodeFile.
+	LineID string `json:"LineId,omitempty"`
+	// RelatedToLine is the ID of a line that when hidden, also hides this line
+	RelatedToLine string `json:"RelatedToLine,omitempty"`
+	// Tokens that construct a line in API review. This is the only required field.
+	Tokens []ReviewToken `json:"Tokens"`
+}
+
+// MarshalJSON ensures the Tokens field isn't marshalled as null, as required
+// by the schema
+func (r ReviewLine) MarshalJSON() ([]byte, error) {
+	type Alias ReviewLine
+	if r.Tokens == nil {
+		r.Tokens = []ReviewToken{}
+	}
+	return json.Marshal((Alias)(r))
 }
 
-// PackageReview ...
-type PackageReview struct {
-	Diagnostics []Diagnostic `json:"Diagnostics,omitempty"`
-	Language    string       `json:"Language,omitempty"`
-	Name        string       `json:"Name,omitempty"`
-	Tokens      []Token      `json:"Tokens,omitempty"`
-	Navigation  []Navigation `json:"Navigation,omitempty"`
-	PackageName string       `json:"PackageName,omitempty"`
+// ReviewToken corresponds to an individual token within a ReviewLine
+type ReviewToken struct {
+	HasPrefixSpace  bool `json:"HasPrefixSpace,omitempty"`
```

Comment:
From some comments, I'm assuming this field is optional. What does `omitempty` do for the zero-value (presumably nothing)? Would it help if the optional fields are pointer-to-type?

Reply:
`omitempty` prevents marshalling the zero value, so we never output `"HasPrefixSpace": false`. I don't see a case in which a pointer value would help